src/apps/vpn/models/recommendedlocationmodel.cpp

```diff
@@ -118,10 +131,13 @@ QVariant RecommendedLocationModel::data(const QModelIndex& index,
   if (!index.isValid() || index.row() >= m_recommendedCities.length()) {
     return QVariant();
   }
-
+  QPointer<ServerCity> city = m_recommendedCities.at(index.row());
   switch (role) {
     case CityRole:
-      return QVariant::fromValue(m_recommendedCities.at(index.row()));
+      if (city.isNull()) {
+        return QVariant();
+      }
+      return QVariant::fromValue(city.data());
```

Comment:
Just checking that will be no reentrancy during the Qt binding evaluation after this line that can potentially cause a ServerCountryModel change signal to be emitted. If so, that can cause the same dangling pointer crash. I'd expect the Qt binding evaluation to occur without such re-entrancy, but I'm checking because I'm new to this framework.

Reply:
Thanks for confirming.